pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -235,10 +235,68 @@ var logger *logrus.Logger
 var log *base.LogObject
 var zedcloudCtx *zedcloud.ZedCloudContext
 
+type infoDest uint
+
+// destination types, where info should be sent
+const (
+	ControllerDest  infoDest = 1
+	LocalServerDest          = 2
+	LOCDest                  = 4
+	AllDest                  = ControllerDest | LocalServerDest | LOCDest
+)
+
+func ignoreErrFlags(dest infoDest, ignoreErr infoDest) zedcloud.DeferredItemFlags {
+	flags := zedcloud.DeferredItem_NoFlags
+	if ignoreErr&dest != 0 {
+		// Ignore errors so that the item be handled with highest priority
+		// out-of-band and not be affected by stalled requests in the normal
+		// queue. See the getDeferredPriorityFunctions() for details how the
+		// @DeferredItem_IgnoreError is handled.
+		flags |= zedcloud.DeferredItem_IgnoreError
+	}
+	return flags
+}
+
+func createDeferredItemsByDest(ctx *zedagentContext, dest infoDest,
```

Comment:
Would be helpful to have a comment saying what this function does. Returns zero or more deferred items based on the destination.